agent-any/src/AnyAgentComplex.cc

```diff
@@ -432,7 +432,20 @@ AnyAgent::parseData (char const *&line, const YCPValue & syntax, bool optional)
     if ((line == 0) || (*line == 0))
 	line = getLine ();
     if (line == 0)
+    {
+        // handle special terms that do not need data like optional data
+        if (syntax->valuetype() == YT_TERM)
+        {
+	    YCPTerm term = syntax->asTerm ();
+	    const string s = term->name ();
+            if (s == "Optional" || s == "Skip")
```

Comment:
Match returns previous match, also do not need next line. Documentation suggest that it can be used later e.g. for `Name` element.
I try to fix all all similar problems, so I check documentation and try to behave correctly.